src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -0,0 +1,861 @@
+        activeRegionEvaluationGenotyperEngine.setLogger(logger);
+    }
+
+    private void initializeReferenceReader() {
+        try {
+            // fasta reference reader to supplement the edges of the reference sequence
+            referenceReader = new CachingIndexedFastaSequenceFile(new File(reference));
+        } catch( FileNotFoundException e ) {
+            throw new UserException.CouldNotReadInputFile(new File(reference), e);
+        }
+    }
+
+    private void initializeAssemblyEngine() {
+        // create and setup the assembler
+        assemblyEngine = new ReadThreadingAssembler(hcArgs.assemblerArgs.maxNumHaplotypesInPopulation, hcArgs.assemblerArgs.kmerSizes, hcArgs.assemblerArgs.dontIncreaseKmerSizesForCycles, hcArgs.assemblerArgs.allowNonUniqueKmersInRef, hcArgs.assemblerArgs.numPruningSamples);
+
+        assemblyEngine.setErrorCorrectKmers(hcArgs.assemblerArgs.errorCorrectKmers);
+        assemblyEngine.setPruneFactor(hcArgs.assemblerArgs.MIN_PRUNE_FACTOR);
+        assemblyEngine.setDebug(hcArgs.DEBUG);
+        assemblyEngine.setDebugGraphTransformations(hcArgs.assemblerArgs.debugGraphTransformations);
+        assemblyEngine.setRecoverDanglingBranches(!hcArgs.assemblerArgs.doNotRecoverDanglingBranches);
+        assemblyEngine.setMinDanglingBranchLength(hcArgs.assemblerArgs.minDanglingBranchLength);
+        assemblyEngine.setMinBaseQualityToUseInAssembly(hcArgs.MIN_BASE_QUALTY_SCORE);
+
+        if ( hcArgs.assemblerArgs.graphOutput != null ) {
+            assemblyEngine.setGraphWriter(new File(hcArgs.assemblerArgs.graphOutput));
+        }
+    }
+
+    /**
+     * Instantiates the appropriate likelihood calculation engine.
+     *
+     * @return never {@code null}.
+     */
+    private ReadLikelihoodCalculationEngine createLikelihoodCalculationEngine() {
+        switch ( hcArgs.likelihoodEngineImplementation) {
+            case PairHMM:
+                return new PairHMMLikelihoodCalculationEngine((byte) hcArgs.likelihoodArgs.gcpHMM, hcArgs.likelihoodArgs.pairHMM, log10GlobalReadMismappingRate, hcArgs.pcrErrorModel);
+            case Random:
+                return new RandomLikelihoodCalculationEngine();
+            default:
+                //Note: we do not include in the error message list as it is of no grand public interest.
+                throw new UserException("Unsupported likelihood calculation engine '" + likelihoodCalculationEngine +
+                                        "'. Please use one of the following instead: 'PairHMM' or 'GraphBased'.");
+        }
+    }
+
+    /**
+     * Create the default read filter for use with the HaplotypeCaller
+     *
+     * @param hcArgs HaplotypeCaller arguments
+     * @param header reads header
+     * @return the default read filter for use with the HaplotypeCaller
+     */
+    public static CountingReadFilter makeStandardHCReadFilter( final HaplotypeCallerArgumentCollection hcArgs, final SAMFileHeader header ) {
```

Comment:
It's easier for the client to pass in the whole arg collection without thinking about which parts need to be extracted (plus this method may use more arguments in the future).